dipy/reconst/multi_voxel.py

```diff
@@ -5,6 +5,215 @@
+    shm_as_ndarray(result)[:] = array1d
+
+    return result
+
+
+def parallel_fit_worker(arguments):
+    """
+    Each pool process calls this worker.
+    Fit model on chunks
+
+    Parameters
+    -----------
+    arguments: tuple
+        tuple should contains a model and
+        list of indexes
+
+    Returns
+    --------
+    result: list of tuple
+        return a list of tuple(voxel index, model fitted instance)
+    """
+    model, input_queue, args, kwargs = arguments
+    return [(idx, model.fit(shared_arr[idx], args, kwargs)) for idx in input_queue]
+
+
+def _init_parallel_fit_worker(arr_to_populate, shape):
+    """
+    Each pool process calls this initializer.
+    Load the array to be populated into
+    that process's global namespace
+
+    Parameters
+    -----------
+    arr_to_populate: multiprocessing.Array
+        shared array
+    shape: tuple
+        tuple of array dimensions
+    """
+    global shared_arr
+    with warnings.catch_warnings():
+        warnings.simplefilter('ignore', RuntimeWarning)
+        shared_arr = np.ctypeslib.as_array(arr_to_populate)
+        shared_arr = shared_arr.reshape(shape)
+
+
+def parallel_voxel_fit(single_voxel_fit):
+    """
+    Wraps single_voxel_fit method to turn a model into a parallel multi voxel model.
+    Use this decorator on the fit method of your model to take advantage of the
+    MultiVoxelFit.
+
+    Parameters
+    -----------
+    single_voxel_fit : callable
+        Should have a signature like: model [self when a model method is being
+        wrapped], data [single voxel data].
+
+    Returns
+    --------
+    multi_voxel_fit_function : callable
+
+    Examples:
+    ---------
+    >>> import numpy as np
+    >>> from dipy.reconst.base import ReconstModel, ReconstFit
+    >>> class Model(ReconstModel):
+    ...     @parallel_voxel_fit
+    ...     def fit(self, single_voxel_data):
+    ...         return ReconstFit(self, single_voxel_data.sum())
+    >>> model = Model(None)
+    >>> data = np.random.random((2, 3, 4, 5))
+    >>> fit = model.fit(data)
+    >>> np.allclose(fit.data, data.sum(-1))
+    True
+    """
+
+    def new_fit(model, data, mask=None, *args, **kwargs):
+        """Fit method in parallel for every voxel in data """
+        if data.ndim == 1:
+            return single_voxel_fit(model, data, *args, **kwargs)
+        if mask is None:
+            mask = np.ones(data.shape[:-1], bool)
+        elif mask.shape != data.shape[:-1]:
+            raise ValueError("mask and data shape do not match")
+
+        # Get number of processes
+        nb_processes = int(kwargs['nb_processes']) if 'nb_processes' in kwargs else cpu_count()
+        nb_processes = cpu_count() if nb_processes < 1 else nb_processes
+
+        if nb_processes == 1:
+            return single_voxel_fit(model, data, *args, **kwargs)
+
+        # Get non null index from mask
+        indexes = np.argwhere(mask)
+        # convert indexes to tuple
```

Comment:
You sometime start with a

    # Majuscule
    # miniscule

Could you please keep it standard? Here and elsewhere.